pkg/kubelet/kubelet_pods.go

```diff
@@ -296,13 +296,24 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 		}
 
 		var (
-			hostPath string
-			image    *runtimeapi.ImageSpec
-			err      error
+			hostPath     string
+			image        *runtimeapi.ImageSpec
+			imageSubPath string
+			err          error
 		)
 
+		subPath := mount.SubPath
+		if mount.SubPathExpr != "" {
+			subPath, err = kubecontainer.ExpandContainerVolumeMounts(mount, expandEnvs)
+
+			if err != nil {
+				return nil, cleanupAction, err
+			}
+		}
+
```

Comment:
Do we need to validate subpath for image volume like we do for other volumes?

```
if utilfs.IsAbs(subPath) {
	return nil, cleanupAction, fmt.Errorf("error SubPath `%s` must not be an absolute path", subPath)
}

err = volumevalidation.ValidatePathNoBacksteps(subPath)
if err != nil {
	return nil, cleanupAction, fmt.Errorf("unable to provision SubPath `%s`: %w", subPath, err)
}
```

Reply:
I'm not even sure if we need that validation at all. We already do those in 

https://github.com/kubernetes/kubernetes/blob/87fcae2bc765e4f752bcf7dfbd0c57f75ec751a3/pkg/apis/core/validation/validation.go#L2950-L2960

Reply:
I'd say better safe than sorry, moved the validation down to also apply to the image volume part.

Reply:
> I'm not even sure if we need that validation at all

```
// validatePathNoBacksteps makes sure the targetPath does not have any `..` path elements when split
//
// This assumes the OS of the apiserver and the nodes are the same. The same check should be done
// on the node to ensure there are no backsteps.
func validatePathNoBacksteps(targetPath string, fldPath *field.Path) field.ErrorList {
```

Reply:
I found the comment in the link you provided.